src/hotspot/share/runtime/arguments.cpp

```diff
@@ -543,19 +543,7 @@ static SpecialFlag const special_jvm_flags[] = {
   { "TLABStats",                    JDK_Version::jdk(12), JDK_Version::undefined(), JDK_Version::undefined() },
 
   // -------------- Obsolete Flags - sorted by expired_in --------------
-  { "CriticalJNINatives",           JDK_Version::jdk(16), JDK_Version::jdk(18), JDK_Version::jdk(19) },
-  { "InlineFrequencyCount",         JDK_Version::undefined(), JDK_Version::jdk(18), JDK_Version::jdk(19) },
-  { "G1RSetRegionEntries",          JDK_Version::undefined(), JDK_Version::jdk(18), JDK_Version::jdk(19) },
-  { "G1RSetSparseRegionEntries",    JDK_Version::undefined(), JDK_Version::jdk(18), JDK_Version::jdk(19) },
-  { "AlwaysLockClassLoader",        JDK_Version::jdk(17), JDK_Version::jdk(18), JDK_Version::jdk(19) },
-  { "UseBiasedLocking",             JDK_Version::jdk(15), JDK_Version::jdk(18), JDK_Version::jdk(19) },
```

Comment:
There are a number of servers with start scripts that specify -XX:-UseBiasedLocking. The option may be obsolete and I wonder if better to leave it as obsolete until there is more uptake of JDK 17 which does not need to disable it explicitly.

Reply:
I don't understand the relevance of "uptake of JDK 17" - this change only affects 19. Anyone on 17 will get the deprecation warning as we deliberately deferred obsoleting the flag until JDK 18.

Reply:
It means the start scripts won't be usable with JDK 19+ builds so it complicates testing with those servers. Once these servers get to JDK 17 (actually JDK 15+) then they should be able to drop this option. I think the more general point is that we could be less aggressive about removing obsolete VM options when we know they are being used. In the past, the PermGen options were left as obsolete options for a long time because there were widely used.

Reply:
The roadmap for the deprecation, obsoletion and expiration(removal) of UseBiasedLocking has already been extensively discussed and agreed upon - ref JDK-8256253. Anyone who decides they want to try JDK 19 and have not yet updated a startup script will now be forced to do so - I don't see that as a bad thing. Anyone who has been ignoring the obsoletion warning in 18 testing is just as likely to keep ignoring it in 19 so I don't think we need to keep kicking this can down the road.

Reply:
> The roadmap for the deprecation, obsoletion and expiration(removal) of UseBiasedLocking has already been extensively discussed and agreed upon - ref JDK-8256253. Anyone who decides they want to try JDK 19 and have not yet updated a startup script will now be forced to do so - I don't see that as a bad thing. Anyone who has been ignoring the obsoletion warning in 18 testing is just as likely to keep ignoring it in 19 so I don't think we need to keep kicking this can down the road.

The roadmap is clear, no issue there. I'm just saying that aggressive removal of obsolete options that are used in important servers is a PITA for testing. It's important to be able to run as many existing applications as possible with the JDK 19 EA builds, we just make it harder when it requires digging into scripts and configuration to remove options that would prevent the VM from starting.

Reply:
Sorry but unless there are issues filed to have those scripts updated then what you say will apply the same in 6 months time. And if there are issues filed then they should be fixed now. So I don't see any point in changing the roadmap.